doc/developers/contributing.rst

```diff
@@ -268,100 +253,118 @@ If any of the above seems like magic to you, then look up the `Git documentation
 <http://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html>`_ on the
 web, or ask a friend or another contributor for help.
 
-If some conflicts arise between your branch and the ``master`` branch, you need
-to merge ``master``. For that, you first need to fetch the ``upstream``, and
-then merge its ``master`` into your branch::
+In order to keep your local branch synchronized with the latest
+changes of the main scikit-learn repository, you will need to add
+the ``upstream`` remote::
 
-  $ git fetch upstream
-  $ git merge upstream/master
+    $ git remote add upstream https://github.com/scikit-learn/scikit-learn.git
 
-Subsequently, you need to solve the conflicts. You can refer to the `Git
-documentation related to resolving merge conflict using the command line
-<https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/>`_.
+You can now synchronize your local branch with::
 
-.. note::
+    $ git fetch upstream
+    $ git merge upstream/master
 
-   In the past, the policy to resolve conflicts was to rebase your branch on
-   ``master``. GitHub interface deals with merging ``master`` better than in
-   the past.
+Subsequently, you might need to solve the conflicts. You can refer to the `Git
+documentation related to resolving merge conflict using the command line
+<https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/>`_.
```

Comment:
it would make me happier if we also add a link to `https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging`. I feel like this book is underappreciated and it actually helps people if they read it, or some of it.

Reply:
I love the book too, we have a link a few lines above (that I simplified and put below, now)